### PR TITLE
Only append comments if actually saved

### PR DESCRIPTION
**Pull request by samchrisinger**
# Purpose

There was a general lack of user feedback when adding/saving comments.

# Changes

- only add a comment to the UI once saved sucessfully
- block the commenting UI while a comment is saving

### CODE REVIEW

**Comment by sloria** [comment]
@samchrisinger Can you please look into the failing JS test?

File: website/static/js/registrationUtils.js [hunk 1]
```diff
@@ -291,16 +291,22 @@ var Question = function(questionSchema, data) {
...
-    self.comments.push(comment);
-    self.nextComment('');
-    save();
+
+    var $comments = $(event.target).closest('.registration-editor-comments');
+    $osf.block('Saving...', $comments);
+    return save()
+        .always($osf.unblock.bind(null, $comments))
+        .done(function () {
+            self.comments.push(comment);
```
Comment (sloria) [inline website/static/js/registrationUtils.js]: Why was this moved to the `done` callback? Optimistic updates--e.g. adding a comment without having to wait for a HTTP request to finish--is better UX, so long as errors are handled gracefully.

Comment (samchrisinger) [inline website/static/js/registrationUtils.js]: @sloria this came about based on feedback from QA. There was a general lack of feedback when commenting, and blocking the UI while saving was an easy addition.

Error handling for the comment UI could get awkward-- type a comment, see it get added, then a red alert appears and the comment is deleted? Or moved back to a pending state?

Comment (sloria) [inline website/static/js/registrationUtils.js]: > Error handling for the comment UI could get awkward-- type a comment, see it get added, then a red alert appears and the comment is deleted? Or moved back to a pending state?

I'd think it would be moved back to a pending state. Not that it matters so much. The overwhelmingly common case will be that the request succeeds, so better to optimize the UI for that case.

All that said, this PR is an improvement to the status quo, so I'll stop picking nits. =)
